lib/tasks/pipeline.js

```diff
@@ -93,11 +92,29 @@ module.exports = Task.extend({
 
     var pluginNameRegex = /^(ember\-cli\-deploy\-)(.*)$/;
     var name            = addon.name.match(pluginNameRegex)[2];
-    if (configuredPluginList && configuredPluginList.indexOf(name) === -1) {
-      return;
-    }
 
-    var deployPlugin = addon.createDeployPlugin({name: name});
+    var configuredNameRegex = /(.*?):(.*)/;
+    var configuredPluginHash = Object.create(null);
+    (deployConfig.plugins || []).forEach(function (pluginName) {
+      var split = pluginName.split(':');
+      if(split[0] === name) {
+        if(split[1]) {
+          configuredPluginHash[pluginName.match(configuredNameRegex)[2]] = split[0];
```

Comment:
JavaScript `.split()` accepts a second parameter giving the maximum number of splits to perform. We could pass `2` to get the behavior you described (though it seems like not a great idea for end-users to do this in the first place).